### DISCUSSION TITLE
Can projects be limited to teams?

### DISCUSSION THREAD

**Discussion by nigellh** [DependencyTrack/dependency-track#2851] [General]
Some of the DT Projects that we create need to have a restricted viewing list. At the moment it means that we have to use a separate server to do this. 

It would be good to limit the scope of who can see some projects to a specific set of users.

Has any thought been given to this?

**Comment by Trakeur** [comment]
Hi,
Check under administration -> access management -> Porfolio access control. That's probably what you're looking for.

**Comment by nigellh** [comment]
Possibly and perhaps even probably. I cannot see when I create a project to assign it to a team. If I just create a project, it could look as if I do not go into portfolio access control that no-one can see the project until it is added to a team in there.

Sort of means that only certain people will be able to create projects.

Or is it that you create a project and everyone can see it until it gets added to the portfolio access control team? Then people can see the project for the period of time that it is not assigned to a portfolio access control team. In that case, you might have to give it a dummy name until it has been added and then you can fix it.

Will try some experiments.

**Comment by nigellh** [comment]
Sadly that did not work.

Created a project, created a new team and added that project under the portfolio access control for that team.

Checked that someone not in that team could see it and they could.

Well it does say that it is 'beta'.

**Reply by Trakeur** [reply]
Did you try to restart the dependency track server ?

**Reply by nigellh** [reply]
I am guessing that you are suggesting that this is for turning on the portfolio access control and not for adding each project!

**Comment by nigellh** [comment]
Sadly, my concern is confirmed.

With portfolio access control turned on and DT restarted. None of the projects are visible if they have not been added to a team you are a part of.. 

I can create a project, but then I cannot see it as it is not added to the team I am a part of at the point of creation.

This could be resolved if the ability to add it to a team was part of the project creation. Otherwise, people will be creating projects they can do nothing with until someone with the correct authority adds it to the team it is needed in.

You could give the team control of 'access management' but that gives them the ability to do a lot of things that they should not have access to.

When a project is created, then it needs to be added to a team.

**Reply by Trakeur** [reply]
Yes, I know what you mean, but this can be a problem if you want the various users to be autonomous on their project without the need for someone to grant them access.
From another point of view, you could say that it's normal for a normal user not to have the rights to create a project and add it to his team.
I don't know the inner workings of this feature, but you can open an issue so that the problem can perhaps be dealt with.

**Comment by Trakeur** [comment]
Nevertheless, I believe they're already working on a refactoring of the permission system, which is not totally adapted at the moment, it has to be said.

**Comment by nigellh** [comment]
Turns out there is already an issue for this - https://github.com/DependencyTrack/dependency-track/issues/2636

Thanks, for you help and advice.
